code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessagingEventHandlerUtils.kt

```diff
@@ -0,0 +1,27 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.messaging
+
+import com.adobe.marketing.mobile.MessagingEdgeEventType
+
+/**
+ * Utility class to handle content card event tracking.
+ */
+internal class MessagingEventHandlerUtils {
+    companion object {
+        @JvmStatic
+        internal fun track(propositionId: String, interaction: String?, eventType: MessagingEdgeEventType?) {
+            val contentCardSchemaData = ContentCardMapper.instance.getContentCardSchemaDataForPropositionId(propositionId)
+            contentCardSchemaData?.track(interaction, eventType)
+        }
+    }
+}
```

Comment:
Why not have this in the companion of MessagingEventHandler as a internal method?

Reply:
My bad, I’ll move it inside